🔧 chore: improve Docker Compose examples

saiqulhaq opened the pull request:
To resolve https://github.com/Strech/sidekiq-prometheus-exporter/issues/78

✨ feat: add multi-version support for Sidekiq monitoring setup
🔧 chore: improve Docker configurations and file organization

The changes introduce comprehensive support for monitoring different Sidekiq
versions (5.x through 8.x) with proper metrics collection and visualization.
Key improvements include:

1. Added version-specific Dockerfiles for both metrics and Sidekiq services
2. Standardized Docker configurations with proper working directories
3. Added Grafana dashboard configurations for each Sidekiq version
4. Implemented automatic dashboard provisioning
5. Added proper version-specific gem dependencies
6. Improved file paths and volume mappings in docker-compose
7. Added Grafana environment variables for better customization

These changes make it easier to monitor Sidekiq instances across different
versions while maintaining consistency in the monitoring setup.

saiqulhaq commented:
I have tested all images locally. The metric endpoint works
I pushed the images to Docker Hub if you want to try
https://hub.docker.com/r/saiqulhaq/sidekiq-prometheus-exporter/tags

Please explain what you expect with `Update Graphana board with new metrics` in the https://github.com/Strech/sidekiq-prometheus-exporter/issues/78 @Strech?

Strech commented:
Hey @saiqulhaq thanks for the effort!

The problem is that the title of #78 is aimed to add examples for Docker files in [examples/](https://github.com/Strech/sidekiq-prometheus-exporter/tree/master/examples) folder (see title). There is no need in other Docker files in `docker/` folder.

In the same `examples/` folder you can find Grafana setup which requires some refreshing

saiqulhaq commented:
No worries @Strech, I made this PR with very little effort, made with vibe coding 😅
I will update the PR later
Thanks for your feedback

saiqulhaq commented:
I just updated this PR

![image](https://github.com/user-attachments/assets/67c376d2-4a4d-49f9-9d0e-d96e57f8489c)
Sidekiq 5

![image](https://github.com/user-attachments/assets/ff31fd60-05b4-4061-8c09-102c5c7c69cd)
Sidekiq 6

![image](https://github.com/user-attachments/assets/bf70286b-d03a-4b8f-8a49-cdaaac6f6375)
Sidekiq 7

Sidekiq 8 is working fine. It was the first test I ran, but I forgot to take any screenshots.

saiqulhaq commented:
![localhost_3000_d_sidekiq8x_sidekiq-8-x-dashboard_orgId=1 refresh=5s](https://github.com/user-attachments/assets/d4446a0f-7646-4b0a-8180-b5fad5834a76)

saiqulhaq commented:
![image](https://github.com/user-attachments/assets/4b9614af-972c-4e97-aec6-308abf2676e1)

saiqulhaq committed:
✨ feat: add multi-version support for Sidekiq monitoring setup

🔧 chore: improve Docker configurations and file organization

The changes introduce comprehensive support for monitoring different Sidekiq
versions (5.x through 8.x) with proper metrics collection and visualization.
Key improvements include:

1. Added version-specific Dockerfiles for both metrics and Sidekiq services
2. Standardized Docker configurations with proper working directories
3. Added Grafana dashboard configurations for each Sidekiq version
4. Implemented automatic dashboard provisioning
5. Added proper version-specific gem dependencies
6. Improved file paths and volume mappings in docker-compose
7. Added Grafana environment variables for better customization

These changes make it easier to monitor Sidekiq instances across different
versions while maintaining consistency in the monitoring setup.

Strech commented:
Hey @saiqulhaq I'm afraid that it's a wasting of your time, without a good description of what should be done, the vibe coding is not doing any good for you and it's partially my fault.

I've updated the description of the #78 with steps and details you need to actually help me. Thanks for your patience.

saiqulhaq commented:
thanks for clarifying the issue detail @Strech 

in this PR, I added the suffix requirement in metrics and sidekiq folders. so, only the new metric that is not done yet

miry commented on examples/metrics/Dockerfile.sidekiq-5x:
```diff
@@ -0,0 +1,16 @@
+FROM ruby:2.6.10-alpine3.15
```
Is it not too old?

saiqulhaq commented:
hi @Strech 
any reason why did you close my PR?

Strech commented:
Sorry about that, but I don't have time to go over all the unnecessary changes vibe-coding introduced, from dangling whitespaces, to some structural changes.

It might feel unfair, but I just don't have capacity to review PR generated by AI. I do have AI assistents, like Cursor and so on, but I don't want to spend a second reviewing tons of AI-generated crap.

If you ask me what could be different? I would rather accept personal effort of any minimal changes which are moving project to be better shape, not just to close some issue or tick the TODO box.
